src/Draco.Fuzzer/Testing/ComponentFuzzer.cs

```diff
@@ -1,23 +1,33 @@
-using System.Runtime.CompilerServices;
 using Draco.Fuzzer.Testing.Generators;
 
 namespace Draco.Fuzzer.Testing;
 
 internal abstract class ComponentFuzzer<T>
 {
     private readonly List<(string input, Exception ex)> errors = new();
-
+    private string currentInput = string.Empty;
+    private string path = "lastinputlog.txt";
+    private bool shouldStop = false;
     protected readonly IInputGenerator<T> generator;
 
     public ComponentFuzzer(IInputGenerator<T> generator)
     {
         this.generator = generator;
         Console.CancelKeyPress += new ConsoleCancelEventHandler(this.HandleCancel);
+        if (File.Exists(this.path)) File.Delete(this.path);
+        Console.WriteLine($"in case of stack overflow look at this log for the last input: {Path.GetFullPath(this.path)}");
     }
 
-    public void AddError(Exception ex, string input) => this.errors.Add((input, ex));
+    private void AddError(Exception ex) => this.errors.Add((this.currentInput, ex));
+
+    // Note: We are writing the input to the log from the start, because if we get stack overflow, we wouldn't be able to get to the input
+    private void AddInput(string input)
+    {
+        this.currentInput = input;
```

Comment:
Why do we stash the current input? As far as I can see, it's accessible everywhere you'd need it, no need to introduce state.